### PR TITLE
Login signup views

### PR DESCRIPTION
**Pull request by Alishiaw**
created login and signup fragments.

### CODE REVIEW

File: src/main/java/com/codeup/maktrak/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@Controller
+public class UserController {
+    private UserRepository usersRepository;
+    private PasswordEncoder encoder;
+
+    public UserController(UserRepository usersRepository, PasswordEncoder encoder) {
+        this.usersRepository = usersRepository;
+        this.encoder = encoder;
+    }
+
```
Comment (podaho) [inline src/main/java/com/codeup/maktrak/controllers/UserController.java:24]: Need A Path to Login User; right now there is no way for us to post the user login credentials yet. You need a `@PostMapping("/Login")` or something of the sorts here.

File: src/main/resources/templates/fragments/nav.html [hunk 2]
```diff
@@ -26,8 +26,8 @@ <h4>Modal Header</h4>
             </ul>
 
             <ul id="nav-mobile" class="side-nav">
-                <li><a href="#">Login</a></li>
-                <li><a href="#">Register</a></li>
+                <li><a href="#!">Login</a></li>
```
Comment (podaho) [inline src/main/resources/templates/fragments/nav.html:29]: This should be where you activate the modal for Login...

File: src/main/resources/templates/fragments/nav.html [hunk 3]
```diff
@@ -26,8 +26,8 @@ <h4>Modal Header</h4>
...
+                <li><a href="#!">Register</a></li>
```
Comment (podaho) [inline src/main/resources/templates/fragments/nav.html:30]: This is where you activate the Signup modal for creating new user...

File: src/main/resources/templates/fragments/signup-fragment.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                    <input id="lastname" th:field="*{lastname}">
+
+                    <label for="password">Password</label>
+                    <input id="password" type="password" th:field="*{password}" />
+
+                    <label for="passwordConfirm">Confirm Password</label>
+                    <input id="passwordConfirm" type="password" th:field="*{passwordConfirm}" />
+
+
+                    <input value="Sign me up!" type="submit" />
```
Comment (podaho) [inline src/main/resources/templates/fragments/signup-fragment.html:34]: I believe this is the button that activates modals, you might need to check the Docs for this

File: src/main/resources/templates/fragments/login-fragment.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+<body>
+<div th:fragment="login-modal">
+    <div id="login" class="modal fade">
+       <div class="modal-content">
+          <div class="modal-header">
+              <button type="button" class="close" data-dismiss="modal">&times;</button>
+              <h4 class="modal-title">Please Log In</h4>
+          </div>
+           <div class="modal-body">
+             <div th:if="${param.error}">
```
Comment (podaho) [inline src/main/resources/templates/fragments/login-fragment.html:16]: What `param`?

File: src/main/resources/templates/fragments/login-fragment.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+                Invalid Username and Password.
+             </div>
+             <div th:if="${param.logout}">
```
Comment (podaho) [inline src/main/resources/templates/fragments/login-fragment.html:19]: `param` was never defined?

**Review by podaho — CHANGES_REQUESTED** [review]
I have the Users dao that will be pushed to master shortly. You need that to access the DB to make authentication work... Right now you're encoding and push that info to the database, i think. But I don't know for sure because there's no way to test your code. Make sure you are able to see yourself sign up and login. Looks okay right now, you're still missing some stuff though.

**Comment by fmendozaro** [comment]
@Alishiaw Take a look at Po's comments, this could answer a lot of your support requests

File: src/main/java/com/codeup/maktrak/controllers/UserController.java (hunk 1 above)
Comment (MontealegreLuis) [reply src/main/java/com/codeup/maktrak/controllers/UserController.java:24]: There's no need for a `@PostMapping` when the security component is being used, that's one of the things that will be automatic.

File: src/main/resources/templates/fragments/login-fragment.html (hunk 5 above)
Comment (MontealegreLuis) [reply src/main/resources/templates/fragments/login-fragment.html:16]: `param` referes to the values in the query string. It doesn't need to be defined anywhere. It'll be available in any page.

**Pull request closed without merging**